app/gui/src/project-view/components/GraphEditor/CodeMirrorWidgetBase.vue

```diff
@@ -0,0 +1,183 @@
+  content: model.value,
+  placeholder: () => props.placeholder ?? ' ',
+  extensions: [syncExt],
+  readonly: false,
+  contentTestId: props.contentTestId,
+  lineMode: () => props.lineMode ?? 'single',
+})
+watchEffect(() =>
+  setExtraExtensions([
+    highlightStyle(themeWidget),
+    ...(props.lineMode !== 'multi' ? [selectOnMouseFocus] : []),
+    ...(props.extensions ?? []),
+  ]),
+)
+
+const { getText, setText, onTextEdited, onUserAction } = connectSync(editorView)
+watch(model, (text) => setText(text))
+onTextEdited((text) => {
+  editing.edit(props.transformUserInput?.(text) ?? text)
+  emit('textEdited', text)
+})
+onUserAction((text, selection) => emit('userAction', text, selection))
+
+const previousValue = ref<string>()
+const editing = WidgetEditHandler.New(props, {
+  start() {
+    previousValue.value = model.value
+  },
+  cancel() {
+    if (getText() !== model.value) setText(model.value)
+    blurEditor()
+  },
+  pointerdown(event) {
+    if (targetIsOutside(event, editorRoot.value?.$el)) {
+      accepted()
+    }
+    return false
+  },
+  end() {
+    blurEditor()
+  },
+})
+
+function blurEditor() {
+  editorView.contentDOM.blur()
+}
+
+function focusEditor() {
+  editorView.dispatch({ selection: { anchor: 0, head: editorView.state.doc.length } })
+  editorView.focus()
+}
+
+const inputError = useToast.error()
+
+async function accepted() {
+  const text = getText()
+  if (previousValue.value === text) {
+    editing.cancel()
+    return
+  }
+  const result = await handleAccept(text)
+  if (result.ok) {
+    editing.end()
+  } else {
+    inputError.reportError(result.error)
+    editing.cancel()
+  }
+}
+
+function handleAccept(newValue: string) {
+  if (props.onAccepted) {
+    return props.onAccepted(newValue)
+  } else {
+    model.value = newValue
+    return Ok()
+  }
+}
+
+function onEnter(event: KeyboardEvent) {
+  if (props.lineMode == 'single' || !event.shiftKey) {
+    event.stopPropagation()
+    accepted()
+  }
+}
+
+defineExpose({
+  focusEditor,
+  blurEditor,
+  setText,
+})
+</script>
+
+<template>
+  <CodeMirrorRoot
+    ref="editorRoot"
+    class="CodeMirrorWidgetBase widgetApplyPadding"
+    @focusin="editing.start()"
+    @keydown.enter="onEnter"
+    @keydown.tab.stop.capture="accepted"
+    @keydown.up.stop
+    @keydown.down.stop
+    @click.stop
+  />
+</template>
+<style scoped>
+.CodeMirrorWidgetBase {
+  :deep(.cm-content) {
+    caret-color: var(--color-node-text);
+  }
+  &:deep(::selection) {
+    background: var(--color-widget-selection);
+  }
+}
+</style>
+
+<!--suppress CssUnusedSymbol -->
+<style module="themeWidget">
+.comment,
+.lineComment,
+.blockComment,
+.docComment,
+.name,
+.variableName,
+.definition-variableName,
+.literal,
+.string,
+.escape,
+.number,
+.keyword,
+.moduleKeyword,
+.modifier,
+.punctuation,
+.paren,
+.operator,
+.definitionOperator,
+.invalid {
+  color: var(--color-node-text);
+  transition: color 0.2s ease;
+}
+</style>
```

Comment:
This PR removes syntax highlighting for CM widgets, e.g. in Enso Table Expressions.

Reply:
Fixed:
<img width="745" alt="image" src="https://github.com/user-attachments/assets/2cac89e7-92cf-4f7d-9d67-8471bee93cc3" />